src/compile-route.js

```diff
@@ -27,16 +28,48 @@ function getHeaderMatcher (expectedHeaders) {
 		if (!headers) {
 			headers = {};
 		}
+
+		if (headers instanceof Headers) {
+			headers = headers.raw();
+		}
+
 		const lowerCaseHeaders = Object.keys(headers).reduce((obj, k) => {
 			obj[k.toLowerCase()] = headers[k]
 			return obj;
 		}, {});
+
 		return expectation.every(header => {
-			return lowerCaseHeaders[header.key] === header.val;
+			return areHeadersEqual(lowerCaseHeaders, header);
 		})
 	}
 }
 
+function areHeadersEqual (currentHeader, expectedHeader) {
+    const {key, val} = expectedHeader;
+    const currentHeaderValue = (Array.isArray(currentHeader[key])) ? currentHeader[key] : [currentHeader[key]];
+    const expectedHeaderValue = (Array.isArray(val)) ? val : [val];
+
+    if (currentHeaderValue === expectedHeaderValue) {
+		return true;
+    }
+
+    if (!currentHeaderValue || !expectedHeaderValue) {
+		return false;
+	}
+
+    if (currentHeaderValue.length !== expectedHeaderValue.length) {
+		return false;
+    }
+
+    for (let i = 0; i < currentHeaderValue.length; ++i) {
```

Comment:
In my humble opinion, the order of the values (multiple values per single header field) **do** matter. Therefore I decided to do a strict comparison.
According to the [RFC 2616, sec. 4.2](https://tools.ietf.org/html/rfc2616#section-4.2), the order in between both notations (comma separated list vs. multiple headers with identical name) is significant and no interceptor / proxy is allowed to change this order.

>    It MUST be possible to combine the multiple header fields into one
   "field-name: field-value" pair, without changing the semantics of the
   message, by appending each subsequent field-value to the first, each
   separated by a comma. The order in which header fields with the same
   field-name are received is therefore significant to the
   interpretation of the combined field value, and thus a proxy MUST NOT
   change the order of these field values when a message is forwarded.

In assumption of that, I'd recommend **not** sorting before the comparison. What do you think?